### ISSUE TITLE
Offline Files opening-procedure very slow due to check for newer versions

### ISSUE THREAD

**Issue by SHU-red** [nextcloud/android#11514]
### Is your feature request related to a problem? Please describe.
I often use NextCloud to store/share/access files for travelling documents
A very common issue for me is as i want to open up my boarding pass right getting into the plane when its getting hectic
- I already synced the full folder to my android phoe
- I already created a shortcut on launcher to directly open boarding pass
- After opening the file, there comes a very very long loading screen with the message "trying to find newer file" (or something)
- During this very very long phase, the document can not be seen (and in my case, the QR-code can not be accessed)

### Describe the solution you'd like
- As i open up a file, it immediately opens/shows the current version on the phone
- A loading icon/dialog could be shown in a corner or something and check for newer versions more passively/in the background
- During the "open-phase" where the app tries to contact the NC-server, the files are directly available/zoomable/clickable in the latest version on the phone

### Describe alternatives you've considered
- There are no alternatives, because im syncing my files offline to have them available immediately

### Additional context
- none

**Comment by tobiasKaminsky** [comment]
Your solution is a great idea, but only for files that are shown within app.
Check this situation:
- you have a PDF
- it is updated on server, meanwhile you are already in the folder (or as you have, use shortcut)
- open PDF
- files app opens external PDF viewer
- files app checks in background that PDF is outdated
- files app tries to inform user that PDF is outdated -> not possible, as app is not visible

This blocking mechanism is a trade off between speed/UX and making sure that file in question is really up to date.

**Comment by SHU-red** [comment]
I fully understand your point ...

What about this:
- If i use the "On your Device" path to open a file or create a shortcut, it does not try to find a newer version or trying to sync several seconds but opens the local version directly
- If i use the "All Files" path or one of the others, it tries to sync before opening

It feels really strange to me, that i am somewhere offline and i have to wait this long time before a file opens which i did prepare on purpose to acess it

If i have to search through e.g. 5 files because i don't exactly know where to find my information i will have to wait for ever until i see what the file contains

The above mentioned solution (or an alternative) could also be configurable via option

Or what about a switch on the main ui to switch between Only Offline or Online mode?

**Comment by SHU-red** [comment]
Yes I've definitely not been at home
Could this be the case when being in a public WiFi (like on airports)?

And then the app is searching for the configured server IP but the process is slow or something?

**Comment by SHU-red** [comment]
I tried to screen-capture this for you as good as possible.

As Sidenotes for the video:

- My nextcloud is running in docker on my homeserver
- Ive deactivated my wifi so im only connected to the cellular network
- Cellular network is not very good during this test

This is how it lookes and sometimes the "loading-dialogs" appear multiple times and last longer

Hopefully this is helping you?

https://www.dropbox.com/s/g3atf6oym8ek58e/NextCloud_LaggyOfflineFiles.mp4?dl=0

**Comment by maxim-kukushkin** [comment]
I totally agree with the original use case and request. I've actually had a very similar experience. I was at the gate in an airport, offline and I needed to show my boarding pass which I synchronized in advance. However, being slightly distracted I forgot that the file is not open yet. I tapped it in the NC app and saw this dialog, not disappearing for 5-10 seconds, during which I was delaying the queue.

First of all, unfortunately it's not true that the window disappears in 0.5 seconds - it doesn't. That time (above) I was offline, with no network access. I also had the same experience in airplane mode (while trying to open a magazine during the flight). But worst thing is if you're online, but with slow network.

I understand the concern that the user might want the latest version of the file when opening it being online, but I believe this experience requires attention and some proper way of fixing it.

When people do extra action asking the app to synchronize the file for offline use, it's normally on purpose and they need **true** offline experience - tap -> open immediately

As a super quick workaround, I'd suggest adding "Skip" button to the modal window which tries to check for the latest version. In this case, if I don't need the latest file or I know it will take too long and I need the file ASAP, I can simply tap the button

Long term, perhaps there can be a control in the UI which activates "offline" mode (the app can change color as an indication for example).

Another option is to play with file icons. I.e., when the user enters a directory with "offline" files - those files are marked with a special icon indicating "available, but not confirmed". When the file list is then asynchronously fetched from server and the file is confirmed to be up-to-date, the icon changes to the green tick, if it's not up-to-date the icon changes respectively and it can be asynchronously fetched

**However, as a quick fix, I'd add the "Skip" button to the synchronization dialog**

**Comment by joshtrichards** [comment]
Related: #1143
